### PR TITLE
Arch, Alpine, Fedora, RHEL: update to latest system version

### PR DESCRIPTION
**Pull request by ghost**
- Update RHEL version from 9.1 to 9.2, fix broken download link.
- Update Fedora version from 37 to 38
- Update Arch version from 20230401 to 20230503
- Update Alpine version from 3.17 to 3.18
- Update Fedora, RHEL zfs-repo from 2-2 to 2-3
- Fix one typo in NixOS docs
- Add two lines of missing commands in zfs_root_maintenance.rst

Closes #434

@gmelikov

### CODE REVIEW

File: docs/Getting Started/zfs_root_maintenance.rst [hunk 1]
```diff
@@ -140,6 +140,8 @@ and other maintenance operations and is thus currently not supported.
    ::
 
       set -x
+      boot_dataset=$(df -P /boot | tail -n1 | cut -f1 -d' ' || true )
```
Comment (gmelikov) [inline docs/Getting Started/zfs_root_maintenance.rst:143]: Looks like it already is here, line 146 in new variant.

**Comment by ne9z** [comment]
Yes.  But someone who wish to destroy the BE would very likely miss the
step where boot_dataset variable was set, which is currently at the very
beginning of BE creation steps.  So, we mention it again inside the BE
destruction steps to avoid any confusion.

File: docs/Getting Started/zfs_root_maintenance.rst (hunk 1 above)
Comment (gmelikov) [reply docs/Getting Started/zfs_root_maintenance.rst:143]: Then let's just remove second declaration from line 146?

**Review by gmelikov — APPROVED** [review]
Thank you!

**Comment by ne9z** [comment]
Thanks for reviewing and merging this!  I see that the Sphinx build
failed somehow; and the pages were not updated after merging this PR to
master.  Maybe we should trigger a manual rebuild of the GitHub pages.

**Comment by gmelikov** [comment]
@ne9z yep, http://cvsweb.netbsd.org/bsdweb.cgi/~checkout~/src/external/cddl/osnet/dist/cmd/zpool/zpool-features.7?content-type=text/plain&only_with_tag=MAIN... doesn't work now, I'll rerun CI when it will be available.

**Pull request merged**